packages/block/src/header.ts

```diff
@@ -155,7 +156,9 @@ export class BlockHeader {
       mixHash: zeros(32),
       nonce: zeros(8),
       baseFeePerGas: undefined,
-      withdrawalsRoot: undefined,
+      // TODO: Remove this if needed since we expect CL to provide a withdrawals array (even empty)
+      withdrawalsRoot: this._common.isActivatedEIP(4895) ? KECCAK256_RLP : undefined,
```

Comment:
Is this default initialization useful (so: to set to KECCAK256_RLP if 4895 activated + root not set)? Or would an alternative be to throw here?

(absolutely no suggestion, do not know the context enough)

Reply:
This is just a default for when 4895 is active.  My instinct is that we should provide this as a default option for when library consumers are constructing block headers outside of the client context.  In the client, this withdrawals array should never be undefined as we expect the CL client to provide the withdrawals array as part of the execution payload but we don't have anything in our Block/Header constructors that constructs a withdrawalsRoot  at present so we should either provide that in the static constructors or leave this default here.  Otherwise, you get hard to understand errors and it may not be obvious how one generates a `withdrawalsRoot` since it's a trie constructed from the withdrawal transactions included with a block.

Reply:
👍